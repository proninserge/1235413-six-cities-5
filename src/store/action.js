const ActionType = {
  CITY_CHANGE: `CITY_CHANGE`,
  SET_SORT_TYPE: `SET_SORT_TYPE`,
  RESET_SORT_TYPE: `RESET_SORT_TYPE`,
};

const ActionCreator = {
  changeCity: (city) => ({
    type: ActionType.CITY_CHANGE,
    city,
  }),
  setSortType: (sortType) => ({
    type: ActionType.SET_SORT_TYPE,
    sortType,
  }),
  resetSortType: () => ({
    type: ActionType.RESET_SORT_TYPE,
  }),
};

export {ActionType, ActionCreator};
