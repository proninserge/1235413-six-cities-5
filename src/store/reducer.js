import offers from '@/mocks/offers';
import reviews from '@/mocks/reviews';
import {ActionType} from '@store/action';
import {getOffers} from '@/location';

const allCities = new Set([...offers.map((offer) => offer.city)]);
const cities = [...allCities];
const initialCity = cities[0];
const initialOffers = offers.filter((offer) => offer.city === initialCity);

const initialState = {
  currentCity: initialCity,
  offersForCity: initialOffers,
  cities,
  offers,
  reviews,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CITY_CHANGE:
      return Object.assign({}, state, {
        currentCity: action.city
      });
    case ActionType.GET_OFFERS:
      return Object.assign({}, state, {
        offersForCity: getOffers(state.offers, state.currentCity),
      });
    default: return state;
  }
};


export {reducer};
