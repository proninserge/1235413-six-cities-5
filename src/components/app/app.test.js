import renderer from 'react-test-renderer';
import App from './app';
import {ActionCreator} from '@store/action';

import {Provider} from 'react-redux';
import {createStore} from 'redux';
import reducer from '@store/reducers/root-reducer';
import {Router} from 'react-router-dom';
import browserHistory from '@/browser-history';
import leaflet from 'leaflet';

leaflet.map = () => ({
  setView: () => {},
  addLayer: () => {},
});

const store = createStore(reducer);
store.dispatch(ActionCreator.changeCity(`Lafayette`));
const noop = () => {};


describe(`<App /> render`, () => {
  it(`correctly renders App`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={browserHistory}>
              <App
                makeInitialLoad={noop}
              />
            </Router>
          </Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
