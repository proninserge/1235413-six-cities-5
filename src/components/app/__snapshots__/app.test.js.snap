// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render correctly renders App 1`] = `
<h1
  style={
    Object {
      "left": "50%",
      "position": "fixed",
      "textAlign": "center",
      "top": "50%",
      "transform": "translate(-50%, -50%)",
    }
  }
>
  Hold on a sec, we are loading
  <br />
  <small>
    If it takes longer than a few secs,
    <br />
    please, refresh the page.
  </small>
</h1>
`;
